WordPress/gutenberg
# Enable page list to expand in list view

## What?

Closes https://github.com/WordPress/gutenberg/issues/45482

Adds the ability to the page list block to expand in the list view.

## Why?


To reflect on the hierarchical structure that we're representing, particularly when the page list is used inside a navigation block.

## How?


Add a new Page List Item block that can only appear as a chidl of Page List.

## Testing Instructions
- check out this PR
- be sure to have a few pages and some of them to have a parent
- in a post add a Page List block
- open List View 
- notice the page list block can be expanded and all its children appear

## Screenshots or screencast 

https://user-images.githubusercontent.com/107534/202382441-f0750e87-39bd-4fb6-aa67-f583021150f3.mp4


## Known issues:

- [x] While the inner blocks are locked they can be unlocked by the user. The problem is that any edits will not be saved, as the Page List block is always using the hierarchy of pages available, including titles for labels and page URL for links. We need to make an option for locks to be unlock-able or come up with something else.
- [x] The new inner blocks cause some layout issues when the page list block is used in a navigation block
- [x] The post is dirty after the page list renders because the innerblocks are added too late? (fixed in https://github.com/WordPress/gutenberg/pull/45776/commits/f69f59d5746163d9e89d7612c4caf5eac94f817d )
